test/sharness/t0300-docker-image.sh

```diff
@@ -79,9 +79,9 @@ test_expect_success "check that init script configs were applied" '
 '
 
 test_expect_success "simple ipfs add/cat can be run in docker container" '
-  expected="Hello Worlds" &&
-  HASH=$(docker_exec "$DOC_ID" "echo $(cat expected) | ipfs add | cut -d' ' -f2") &&
-  docker_exec "$DOC_ID" "ipfs cat $HASH" >actual &&
+  echo "Hello Worlds" | tr -d "[:cntrl:]" > expected &&
+  HASH=$(docker_exec "$DOC_ID" "echo $(cat expected) | ipfs add -q" | tr -d "[:cntrl:]") &&
+  docker_exec "$DOC_ID" "ipfs cat $HASH" | tr -d "[:cntrl:]" > actual &&
```

Comment:
Two issues were addresses to fix this case:
- replaced `ipfs add | cut -d' ' -f2` which broke quoting (unescaped `'`) with `ipfs add -q`
- added `tr -d "[:cntrl:]"` to all outs we care about because `docker_exec` returns results `\r`

This go me unblocked which is all I was after for now.